### ISSUE TITLE
Command create-project fails on plugins

### ISSUE THREAD

**Issue by janpecha** [composer/composer#10935]
Hello, `create-project` fails if it found `composer.json` with plugin in current working directory.

```
$ composer require frontpack/composer-assets-plugin
```

It creates this `composer.json`:

```
{
    "require": {
        "frontpack/composer-assets-plugin": "^0.11.1"
    },
    "config": {
        "allow-plugins": {
            "frontpack/composer-assets-plugin": true
        }
    }
}
```

Now if I run `create-project` in same directory (I used it on Github Actions) it fails:

```
$ composer create-project janpecha/code-checker _temp/code-checker --no-interaction

In PluginManager.php line 762:
                                                                                                                                                                                           
  frontpack/composer-assets-plugin contains a Composer plugin which is blocked by your allow-plugins config. You may add it to the list if you consider it safe.                           
  You can run "composer config --no-plugins allow-plugins.frontpack/composer-assets-plugin [true|false]" to enable it (true) or disable it explicitly and suppress this exception (false)  
  See https://getcomposer.org/allow-plugins 
```

With option `--no-plugins` it works fine:

```
$ composer create-project janpecha/code-checker _temp/code-checker --no-interaction --no-plugins

Creating a "janpecha/code-checker" project at "_temp/code-checker"
Installing janpecha/code-checker (v1.0.1)
Plugins have been disabled.
  - Installing janpecha/code-checker (v1.0.1): Extracting archive
...
```

**Comment by ihor-sviziev** [comment]
We're having quite a similar issue with command 
```
composer create-project magento/magento-coding-standard magento-coding-standard 16.*
```
on Composer 2.3.7 - worked fine ✅
<img width="1053" alt="image" src="https://user-images.githubusercontent.com/1873745/178316880-22f13431-4282-4106-b77f-5c8f33c8f792.png">


After updating to 2.3.9 - it started to fail: ❌
<img width="1057" alt="image" src="https://user-images.githubusercontent.com/1873745/178316678-ae5cb5b3-2ce5-4457-b365-2db1f226d89d.png">


Looks like adding `--no-plugins` flag does the trick, but I can expect such breaking changes in minor version, not in the patch release.

Looks like the changes caused by https://github.com/composer/composer/pull/10920

**Comment by Seldaek** [comment]
Thanks - that was actually a great find thanks to the plugin security feature, as plugins from the CWD had no business being loaded here when running create-project.

**Comment by secretsayan** [comment]
We are also experiencing the exact same issues in the exact same scenario for `composer validate` command. Moreover, with other commands provided by composer-plugins like [composer](https://github.com/ergebnis/composer-normalize) we are facing the exact same issue, wherein the plugins declared in the CWD is trying to get loaded and disrupting the process.

**Comment by Seldaek** [comment]
Keeping open until I had time to investigate further what @secretsayan reported. In the meantime feel free to try with `composer self-update --snapshot` if the latest version fixes things for you on the create-project side.

**Comment by Seldaek** [comment]
@secretsayan can you describe some more what you are seeing with `validate`? If it fails because you are missing an allow-plugins in the project you are validating that makes sense to me and it is correct behavior IMO. Running validate loads plugins (as do most commands).

composer-normalize also works with the CWD's composer.json AFAIK, so not sure what you were referring to.

Anyway I'll close this again.. If you have a clear repro case for either of these issues please open a new issue with details.

**Comment by Seldaek** [comment]
@secretsayan sorry I spoke too fast - I dug into ergebnis/composer-normalize#865 and realized the problem could be dealt with internally much better. https://github.com/composer/composer/commit/0e59fbb46eee2b89245dc347bfb4d6efbc6f2591 should resolve your issues with validate command as well as normalize.

**Comment by secretsayan** [comment]
@Seldaek I have checked with Composer 2.3.10 and I found that `composer validate` is  fixed now. 

But issue still remains with `composer normalize`.
